service/src/main/java/bio/terra/tanagra/db/ActivityLogDao.java

```diff
@@ -0,0 +1,240 @@
+      LOGGER.debug("CREATE activity log resource: {}", sql);
+      List<MapSqlParameterSource> resourceParamSets = new ArrayList<>();
+      for (int i = 0; i < activityLog.getResources().size(); i++) {
+        ActivityLogResource resource = activityLog.getResources().get(i);
+        resourceParamSets.add(
+            new MapSqlParameterSource()
+                .addValue("activity_log_id", activityLog.getId())
+                .addValue("stable_index", i)
+                .addValue("resource_type", resource.getType().name())
+                .addValue("study_id", resource.getStudyId())
+                .addValue("cohort_id", resource.getCohortId())
+                .addValue("cohort_revision_id", resource.getCohortRevisionId())
+                .addValue("review_id", resource.getReviewId()));
+      }
+      rowsAffected =
+          Arrays.stream(
+                  jdbcTemplate.batchUpdate(
+                      sql, resourceParamSets.toArray(new MapSqlParameterSource[0])))
+              .sum();
+      LOGGER.debug("CREATE activity log resource rowsAffected = {}", rowsAffected);
+    }
+  }
+
+  @ReadTransaction
+  public List<ActivityLog> getAllActivityLogs(
+      int offset,
+      int limit,
+      String userEmailFilter,
+      boolean userEmailExactMatch,
+      ActivityLog.Type activityTypeFilter,
+      ActivityLogResource.Type resourceTypeFilter) {
+    MapSqlParameterSource params =
+        new MapSqlParameterSource().addValue("offset", offset).addValue("limit", limit);
+    String filterSql =
+        renderSqlForActivityLogFilter(
+            userEmailFilter, userEmailExactMatch, activityTypeFilter, resourceTypeFilter, params);
+    String sql =
+        ACTIVITY_LOG_SELECT_SQL
+            + (filterSql.isEmpty() ? "" : " WHERE " + filterSql)
+            + " ORDER BY logged DESC LIMIT :limit OFFSET :offset";
+    LOGGER.debug("GET all activity logs: {}", sql);
+    List<ActivityLog> activityLogs = getActivityLogsHelper(sql, params);
+    LOGGER.debug("GET ALL activity logs numFound = {}", activityLogs.size());
+    return activityLogs;
+  }
+
+  /** Convert an activity log filter object into a SQL WHERE clause. */
+  private String renderSqlForActivityLogFilter(
+      String userEmailFilter,
+      boolean userEmailExactMatch,
+      ActivityLog.Type activityTypeFilter,
+      ActivityLogResource.Type resourceTypeFilter,
+      MapSqlParameterSource params) {
+    if ((userEmailFilter == null || userEmailFilter.isEmpty())
+        && activityTypeFilter == null
+        && resourceTypeFilter == null) {
+      return "";
+    }
+
+    // Filter on the userEmail, activityType, and/or resourceType.
+    List<String> whereConditions = new ArrayList<>();
+    if (userEmailFilter != null && !userEmailFilter.isEmpty()) {
+      if (userEmailExactMatch) {
+        whereConditions.add("user_email = :user_email_filter");
```

Comment:
Does this need to be sanitized?

Reply:
No, the `MapSqlParameterSource` (see nearby line `params.addValue`) handles escaping all parameters before executing the prepared statement.